functorch/functorch/_src/remat_utils_mincut.py

```diff
@@ -0,0 +1,447 @@
+        break
+
+    env = {}  # map from node in origin to node in dest
+    # create new placeholders in module_dest
+    for node_name in cut_nodes:
+        node = name_to_node_origin[node_name]
+        if node_name in name_to_node_dest:
+            # already has a placeholder for it in dest
+            env[node] = name_to_node_dest[node_name]
+            continue
+        with module_dest.graph.inserting_before(first_node_dest):
+            new_node = module_dest.graph.placeholder(node.name, type_expr=node.type)
+            new_node.meta = copy.copy(node.meta)
+            env[node] = new_node
+
+    # copy internal nodes
+    node_to_copy = get_node_to_copy(non_reachable, cut_nodes)
+    node_to_copy = order_topologically(node_to_copy, module_origin)
+    for node_name in node_to_copy:
+        node = name_to_node_origin[node_name]
+        with module_dest.graph.inserting_before(first_node_dest):
+            new_node = module_dest.graph.node_copy(node, lambda x: env[x])
+            new_node.name = node.name  # use the same name such that node can be referenced back to original graph
+            env[node] = new_node
+            # change the args of nodes in dest to use the new node
+            # e.g. a placeholder is now a node to recompute
+            if node.name in name_to_node_dest:
+                name_to_node_dest[node.name].replace_all_uses_with(new_node)
+
+    # erase unused placeholder nodes and record current active placeholders
+    active_placeholders = []
+    for node in module_dest.graph.nodes:
+        if node.op == "placeholder":
+            if len(node.users) == 0:
+                module_dest.graph.erase_node(node)
+            else:
+                active_placeholders.append(node.name)
+
+    legalize_graph(module_dest)
+    module_dest.graph.eliminate_dead_code()
+    module_dest.graph.lint()
+
+    # change the args of node_pair[1] to match the new placeholders of module_dest
+    origin_placeholder_map = get_name_to_args_map(node_pair[0], module_origin)
+    placeholder_map.update(origin_placeholder_map)
+    for node in module_origin.graph.nodes:
+        if node.op == "output":
+            user_name_to_user_map = get_user_name_to_user_map(node, node_pair[0])
+            placeholder_map.update(user_name_to_user_map)
+
+    new_args = []
+    for name in active_placeholders:
+        if name in name_to_node:  # name is a node in fused graph
+            new_args.append(name_to_node[name])
+        else:  # name is a placeholder in origin's module or dest's module or a newly added input
+            new_args.append(placeholder_map[name])
+    node_pair[1].args = tuple(new_args)
+    fused_graph.recompile()
+
+    fused_graph.graph.eliminate_dead_code()
+    fused_graph.graph.lint()
+    module_dest.recompile()
+
+    # remove the unused output to write less
+    remove_unused_output(fused_graph, module_origin)
+
+
+def find_min_cut(node_pair, node_users_map, fused_graph):
```

Comment:
In general, the doc-blocks put here should describe what it's taking in, as well as what it's returning. Type hints with mypy would be even better :)

Reply:
added some docstring